File: BackEnd/NodeEditor.RestAPI/NodeEditor.Entities/DataInput.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NodeEditor.Entities
{
    public class DataInput:EntityBase
    {
        public DataInputType DataInputType { get; set; }
        public string Label { get; set; }
        public string Name { get; set; }
        public Guid NodeTypeId { get; set; }
        public NodeType NodeType { get; set; }
        public ICollection<DataInputValue>? DataInputValues { get; set; }
    }

    public enum DataInputType
    {
        Number=0,
        Text=1,
        Boolean=2
    }
}

File: FrontEnd/node-editor/src/Models/DataInput.ts
import { NodeType } from "./NodeType";
import { DataInputValue } from "./DataInputValue";
import { EntityBase } from "./EntityBase";

export interface DataInput extends EntityBase {
    dataInputType: DataInputType;
    label: string;
    name: string;
    nodeTypeId: string;
    nodeType?: NodeType;
    dataInputValues: DataInputValue[] | null;
}

export enum DataInputType {
    Number = 0,
    Text = 1,
    Boolean = 2
}